smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -332,10 +333,34 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         );
       } else {
         // if we are double pegging and already control the token, then we are going to need to get the address on this chain
-        tokenAddress = sourceAddressToDestinationAddress[claimData.tokenAddress];
+        tokenAddress = cosmosDenomToDestinationAddress[claimData.cosmosDenom];
       }
-    } else {
-      tokenAddress = claimData.tokenAddress;
+    } else if (isIbcToken(claimData.cosmosDenom)) {
```

Comment:
Do we need to actually make a distinction between a double pegged token and a ibc token on the smart contract side of things? Instead of having a field that says `doublePeg` why not instead just have a field that says `bridgeToken` and offload that logic out of the smart contracts? Both will be bridge tokens be it double pegged or IBC and both will be burned so why handle them differently?

Reply:
I think you are right, now we can use a flag like bridgeToken to replace both doublePeg and isIbcToken judgement. the reason that we handle it differently is the requirement, I think. The old procedure is we add the ibc denom into registry and deploy bridge token for ibc token, and in the cosmosBridge contract, we use the original contract address as the key, but ibc token doesn't have the field.

Reply:
In the sifnode side, we still need doublePeg field since we have different strategy whether to charge the bridge token deployment fee for user.

Reply:
sure but we can move the logic into sifnode or even the relayers to look at the cosmos denom and determine if its a double pegged token or a ibc token right? If its `rowan` its native rowan, if it starts with `ibc/` its an ibc asset else its a double bridged asset.